src/descriptor/mod.rs

```diff
@@ -911,5 +941,374 @@ mod tests {
                 .into_script()
         );
     }
+
+    #[test]
+    fn interpret()
+    {
+        fn setup_keys_sigs(n: usize)
+                           -> ( Vec<PublicKey>, Vec<Vec<u8> >, secp256k1::Message, Secp256k1<All>) {
+            let secp = secp256k1::Secp256k1::new();
+            let msg = secp256k1::Message::from_slice(
+                &b"Yoda: btc, I trust. HODL I must!"[..]
+            ).expect("32 bytes");
+            let mut pks = vec![];
+            let mut sigs = vec![];
+            let mut sk = [0; 32];
+            for i in 1..n+1 {
+                sk[0] = i as u8;
+                sk[1] = (i >> 8) as u8;
+                sk[2] = (i >> 16) as u8;
+
+                let sk = secp256k1::SecretKey::from_slice(&sk[..]).expect("secret key");
+                let pk = PublicKey {
+                    key: secp256k1::PublicKey::from_secret_key(
+                        &secp,
+                        &sk,
+                    ),
+                    compressed: true,
+                };
+                let sig = secp.sign(&msg, &sk);
+                let mut sigser = sig.serialize_der();
+                sigser.push(0x01); // sighash_all
+                pks.push(pk);
+                sigs.push(sigser);
+            }
+            (pks, sigs, msg, secp)
+        }
+
+        let (pks,sigs, msg, secp) = setup_keys_sigs(10);
+
+        //check satisfied case Pk
+        let ms = Miniscript(astelem::AstElem::Pk(pks[0].clone()));
+        let bare = Descriptor::Bare(ms.clone());
+        let mut witness = vec![sigs[0].clone()];
+        let res = bare.interpret(
+            &mut witness,
+            0,
+            &msg,
+            &secp).expect("Interpretation to succeed");
```

Comment:
Style nit: everywhere you have a `)` on the last line of a multiline thing like this, it would be better to have
```
    ...
    0,
    &msg,
    &secp,
).expect("Interpretation to succeed");
```
Note the trailing comma on `secp` and that the `)` gets its own line.

Let me know if you can cite some style guide saying that your way is correct ... I swear the Rust community changes stuff like this every week.